doc/whats_new/v1.4.rst

```diff
@@ -89,6 +89,14 @@ Changelog
   :pr:`13649` by :user:`Samuel Ronsin <samronsin>`, initiated by
   :user:`Patrick O'Reilly <pat-oreilly>`.
 
+:mod:`sklearn.metrics`
+...................
+
+- |Performance| Computing pairwise distances for (CSR x CSR) and (CSR x Dense)
+  datasets is now 1.5x faster by improving the argument passing strategy used
+  in the computation routines in :class:`metrics.DistanceMetric`.
```

Comment:
Nitpick: I think that we generally keep changelog entry short without giving too much technical details.
```suggestion
- |Performance| Computing pairwise distances via :class:`metrics.DistanceMetric` for CSR × CSR,  Dense × CSR, and CSR × Dense datasets is now 1.5x faster.
```

Should we also add another one for estimators of `module.neighbors`?

Reply:
> Should we also add another one for estimators of `module.neighbors`?

I'm not sure -- there are a lot of estimators which at least partly use the `DistanceMetric` backend 
(the following list is puled from https://github.com/scikit-learn/scikit-learn/issues/26329):

```
- NearestNeighbors
- KNeighborsRegressor
- KNeighborsClassifier
- RadiusNeighborsRegressor
- RadiusNeighborsClassifier
- DBSCAN
- OPTICS
- Isomap
- TSNE (self.method != "exact")
- KernelDensity
- AffinityPropagation
- Birch
- MeanShift
- NearestCentroid
```

 I don't quite know where to draw the line for this if we do include it for other estimators specifically. Most `.predict` methods, for example, benefit from this.